test/integration/controller_instance_test.go

```diff
@@ -1088,14 +1082,24 @@ func FlakeTestCreateServiceInstanceWithProvisionFailure(t *testing.T) {
 				URL: "https://fakebroker.com/v2/service_instances/instance_id",
 				Err: TimeoutError("timeout error"),
 			},
-			conditionReason:          "ProvisionedSuccessfully",
-			expectFailCondition:      false,
+			provisionErrorReason:     "ErrorCallingProvision",
 			triggersOrphanMitigation: true,
 		},
 	}
 	for _, tc := range cases {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
+
+			respondWithProvisionSuccess := int32(0)
+			respondWithDeprovisionSuccess := int32(0)
+			blockDeprovision := int32(1)
+
+			// Ensure that broker requests respond successfully after running
+			// the core of the test so that the resource cleanup can proceed.
+			defer atomic.StoreInt32(&respondWithProvisionSuccess, 1)
+			defer atomic.StoreInt32(&respondWithDeprovisionSuccess, 1)
+			defer atomic.StoreInt32(&blockDeprovision, 0)
```

Comment:
not a super fan of atomic ints for what are essentially bools. 
I'm fine with the "throw a big lock around it" solution.

Reply:
It is more code (which is subjectively more difficult to read). I personally don't find it worthwhile to use a mutex to avoid using as integer as a boolean, but I'll make the change nevertheless.

Reply:
You are right about select and chan being more idiomatic, though. I'll use that instead.